scanner/src/main/kotlin/Scanner.kt

```diff
@@ -786,3 +800,20 @@ private fun ProvenanceResolutionResult.filterByVcsPath(): ProvenanceResolutionRe
             File(path).startsWith(packageProvenance?.vcsPath.orEmpty())
         }
     )
+
+/**
+ * Return the VCS path applicable to a (sub-) repository which appears under [repositoryPath] in the source tree of
+ * a package residing in [vcsPath], or null if the subtrees for [repositoryPath] and [vcsPath] are disjoint.
+ */
+private fun getVcsPathForRepositoryOrNull(vcsPath: String, repositoryPath: String): String? {
+    val repoPathFile = File(repositoryPath)
+    val vcsPathFile = File(vcsPath)
+
+    if (repoPathFile.startsWith(vcsPathFile)) {
+        return ""
+    } else if (vcsPathFile.startsWith(repoPathFile)) {
+        return vcsPathFile.relativeTo(repoPathFile).path
+    }
+
+    return null
```

Comment:
At a quick glance, this seem to do the same and is shorter:

    return if (repoPathFile.startsWith(vcsPathFile)) {
        ""
    } else {
        runCatching { vcsPathFile.toRelativeString(repoPathFile) }.getOrNull()
    }